src/v/storage/compaction_reducers.cc

```diff
@@ -265,12 +265,11 @@ ss::future<ss::stop_iteration>
 copy_data_segment_reducer::operator()(model::record_batch&& b) {
```

Comment:
I'd recommend taking this opportunity to change the method to take `b` by value since this is a coroutine now. code is ok, but taking references in coroutines makes them prone to dangling reference bugs (same with the previous commit's changes).

Reply:
Added another commit towards the end for this, thank you.